File: 2.hashmap/1.easy/383. Ransom Note.py
from collections import Counter


def ransomNote(ransomNote, magazine):
    
    ransom_count = Counter(ransomNote)
    magazine_count = Counter(magazine)
    
    

    # Check if each character in the ransom note can be constructed from the magazine
    for char, count in ransom_count.items():
        print(magazine_count[char],count)
        if magazine_count[char] < count:
            return False

    return True



#(ransomNote("a", "b"))         
(ransomNote("aa", "ab"))       
#(ransomNote("aa", "aab"))      

File: javascript/383. Ransom Note.js
function Ransom_Note(ransomNote, magazine){
    ransomNote = ransomNote.split('');
    magazine = magazine.split('');
    
    for (let i =0; i <magazine.length; i++){
            if (!(ransomNote.includes(magazine[i]))){
                return false;
            }
            if (magazine[i] == ransomNote[0]){
                ransomNote.shift();
            }
            if (ransomNote.length === 0){break};
    }
    console.log(ransomNote)
};

let ransomNote = "aab";
let magazine = "aba";
Ransom_Note(ransomNote, magazine)